neps/nep-0413.md

```diff
@@ -0,0 +1,76 @@
+
+If the wallet allows apps to request the user to sign arbitrary data, both attacks would be possible. Thus, some measures are needed to prevent both attacks:
+
+1. To prevent the malicious signing of transactions, a good solution is to make sure that the data to be signed cannot be parsed as a transaction. This is preferred to having a properly formatted transaction with invalid data (such as an invalid method). The precise format of that data should be made a standard, so that, for example, a future change to transaction format will not make it possible to confuse the data for a transaction.
+2. To prevent one app from relaying a signature to a different app, the data to be signed should include the requesting app's identity (e.g. domain name). Again, a well-defined format will facilitate this.
+
+Using an access key restricted in such a way that it cannot send transactions looks like a good temporary solution. It doesn't solve the second problem, though.
+
+Sending the signature in a query string to an arbitrary URL (even within the correct domain) is not secure. Most apps will have some pages that will leak this data in the Referrer header or by some other means. Using URL fragment instead of a query string will improve this (the fragment isn't included in the Referrer), and somehow permitting only a fixed callback URL will improve this even more.
+
+Also, note that NEAR transaction signatures are not plain Ed25519 signatures but Ed25519 signatures of a SHA-256 hash (see [near/nearcore#2835](https://github.com/near/nearcore/issues/2835)). Any protocol that signs anything with NEAR account keys should use the same signature format.
+
+Hence, the final solution we are settling on is to user a restricted payload as shown below.
+
+## Specification
+
+### `verifyOwner`
+
+Signs the message and verifies the owner. Message is not sent to blockchain.
+
+```jsx
+interface VerifyOwnerParams {
+  // The message requested sign. Defaults to `verify owner` string.
+  message?: string;
+   //Account ID used to sign the message. Defaults to the first account
+  signerId?: string;
+  //Public key used to sign the message. Defaults to the public key of the signed in account.
+  publicKey?: PublicKey;
+  // Applicable to browser wallets (e.g. MyNearWallet). This is the callback url once the signing is approved. Defaults to `window.location.href`.
+  callbackUrl?: string;
+  // Applicable to browser wallets (e.g. MyNearWallet) extra data that will be passed to the callback url once the signing is approved.
+  meta?: string;
+}
+
+type VerifyOwnerResponse = Promise<void | utils.key_pair.Signature>;
+
+// Browser wallets won't return the signing outcome as they may need to redirect for signing. For MyNearWallet the outcome is passed to the callback url.
+
+```
+
+## Reference Implementation
+You can find a demo implementation of this standard using the project below:
+[near/wallet-selector#320](https://github.com/near/wallet-selector/pull/320)
```

Comment:
The referenced PR was closed and re-opened as https://github.com/near/wallet-selector/pull/391

However it's also merged, so is it still a `demo implementation`? 🤔

Reply:
Not really. I'll reword that. 
It's only a reference showing the standard being adopted in wallet selector and by wallet developers.